clients/drcachesim/tests/record_filter_unit_tests.cpp

```diff
@@ -287,6 +300,123 @@ process_entries_and_check_result(test_record_filter_t *record_filter,
     return true;
 }
 
+bool
+process_entries_and_check_result_against_ground_truth(
```

Comment:
Please re-use the existing process routine which all the existing tests use.  I don't think there's anything special about this filter that isn't already handled?  If so it should be possible to add it to the existing routine?  That seems much better than this duplication.

Reply:
Done. https://github.com/DynamoRIO/dynamorio/pull/6663#discussion_r1586255377